File: app/src/main/java/ua/com/todd/smsforwading/data/SmsDAO.java
package ua.com.todd.smsforwading.data;

import com.j256.ormlite.dao.BaseDaoImpl;
import com.j256.ormlite.support.ConnectionSource;

import java.sql.SQLException;
import java.util.List;

import ua.com.todd.smsforwading.Sms;

public class SmsDAO extends BaseDaoImpl<Sms, Integer> {

    protected SmsDAO(ConnectionSource connectionSource,
                     Class<Sms> dataClass) throws SQLException {
        super(connectionSource, dataClass);
    }

    public List<Sms> getAllSms() throws SQLException {
        return this.queryForAll();
    }
}

File: app/src/main/kotlin/ua/com/todd/smsforwading/data/SmsDAO.kt
package ua.com.todd.smsforwading.data

import com.j256.ormlite.dao.BaseDaoImpl
import com.j256.ormlite.support.ConnectionSource

import java.sql.SQLException

import ua.com.todd.smsforwading.Sms

public class SmsDAO [throws(javaClass<SQLException>())]
internal (connectionSource: ConnectionSource, dataClass: Class<Sms>) : BaseDaoImpl<Sms, Int>(connectionSource, dataClass) {

    throws(javaClass<SQLException>())
    public fun getAllSms(): List<Sms> {
        return this.queryForAll()
    }
}
